packages/api-plugin-fulfillment-method-shipping-flat-rate/src/mutations/createFlatRateFulfillmentMethod.js

```diff
@@ -0,0 +1,53 @@
+import SimpleSchema from "simpl-schema";
+import Random from "@reactioncommerce/random";
+import ReactionError from "@reactioncommerce/reaction-error";
+import methodSchema from "../util/methodSchema.js";
+
+const inputSchema = new SimpleSchema({
+  method: methodSchema,
+  shopId: String
+});
+
+/**
+ * @method createFlatRateFulfillmentMethodMutation
+ * @summary Creates a flat rate fulfillment method
+ * @param {Object} context - an object containing the per-request state
+ * @param {Object} input - Input (see SimpleSchema)
+ * @returns {Promise<Object>} An object with a `method` property containing the created method
+ */
+export default async function createFlatRateFulfillmentMethodMutation(context, input) {
+  const cleanedInput = inputSchema.clean(input); // add default values and such
+  inputSchema.validate(cleanedInput);
+
+  const { method: inputMethod, shopId } = cleanedInput;
+  const { collections } = context;
+  const { Fulfillment } = collections;
+  const method = { ...inputMethod };
+
+  await context.validatePermissions("reaction:legacy:fulfillmentMethods", "create", { shopId });
+
+  const shippingRecord = await Fulfillment.findOne({ fulfillmentType: "shipping", shopId });
+  if (!shippingRecord) throw new ReactionError("server-error", "Unable to create fulfillment method without defined type");
+
+  method._id = Random.id();
+  // MongoDB schema still uses `enabled` rather than `isEnabled`
+  method.enabled = method.isEnabled;
+  delete method.isEnabled;
```

Comment:
We shouldn't need to do this in a new collection?

Reply:
Ok, so when we have the new collection with enabled instead of isEnabled, we will have to include that in the data migration PR since the old shipping collection had isEnabled.

Reply:
Updated in PR #6614 
https://github.com/reactioncommerce/reaction/pull/6614/commits/da70978d5f566e855a9c59ee4028386b686421f3